scripts/projects.js

```diff
@@ -1,32 +1,61 @@
-var projects = [];
+(function(module){
 
 function Project (opts) {
-  this.title = opts.title;
-  this.category = opts.category;
-  this.developer = opts.developer;
-  this.developerUrl = opts.developerUrl;
-  this.publishedOn = opts.publishedOn;
-  this.body = opts.body;
+  for (keys in opts) {
+    this[keys] = opts[keys];
+  }
 };
 
+Project.all = [];
+
 // Incorporate handlebars.js into code
-Project.prototype.toHtml = function() {
-  var source = $('#project-template').html();
+Project.prototype.toHtml = function(headScriptTemplate) {
+  var source = $(headScriptTemplate).html();
   var template = Handlebars.compile(source);
   var html = template(this);
-
   return html;
 };
 
-myLocalData.sort(function(currentElement, nextElement) {
-  return (new Date(nextElement.publishedOn)) - (new Date(currentElement.publishedOn));
-});
+Project.loadAll = function(dataToPassIn) {
+  dataToPassIn.sort(function(currentElement, nextElement) {
+    return (new Date(nextElement.publishedOn)) - (new Date(currentElement.publishedOn));
+  }).forEach(function(element) {
+    Project.all.push(new Project(element));
+  });
+};
+
+Project.fetchAll = function() {
+  if (localStorage.blogProjects) {
```

Comment:
This check will always fail, because in line 47 you name the localstorage key something different.